### PR TITLE
Updates & fixes for JS inliner

### LINKED ISSUE #1805: eqString is not being inlined
I noticed this in some output:
```js
if (Prelude["=="](Prelude.eqString)(l)("")) {
```
And that should probably be getting inlined.

### PR DESCRIPTION
**Pull request by garyb**
Resolves #1805

Manually tested by checking the output of this with both old and new (moduleified) prelude:

``` purescript
module Main where

import Prelude (Unit, void, pure, (>>>), (#), negate, ($), (<<<), bind, mod, div, (/), mul, (*), sub, (-), add, (+), not, disj, conj, (&&), append, (<>), (>=), (<=), (>), (<), (/=), eq, (==))
import Control.Monad.Eff (Eff)
import Control.Monad.Eff.Console (CONSOLE)

main :: forall e. Eff (console :: CONSOLE | e) Unit
main = do

  void <<< pure $ (true == false)
  void <<< pure $ (1    == 2)
  void <<< pure $ (1.0  == 2.0)
  void <<< pure $ ('a'  == 'b')
  void <<< pure $ ("a"  == "b")

  void <<< pure $ (true `eq` false)
  void <<< pure $ (1    `eq` 2)
  void <<< pure $ (1.0  `eq` 2.0)
  void <<< pure $ ('a'  `eq` 'b')
  void <<< pure $ ("a"  `eq` "b")

  void <<< pure $ (true /= false)
  void <<< pure $ (1    /= 2)
  void <<< pure $ (1.0  /= 2.0)
  void <<< pure $ ('a'  /= 'b')
  void <<< pure $ ("a"  /= "b")

  void <<< pure $ (true < false)
  void <<< pure $ (1    < 2)
  void <<< pure $ (1.0  < 2.0)
  void <<< pure $ ('a'  < 'b')
  void <<< pure $ ("a"  < "b")

  void <<< pure $ (true > false)
  void <<< pure $ (1    > 2)
  void <<< pure $ (1.0  > 2.0)
  void <<< pure $ ('a'  > 'b')
  void <<< pure $ ("a"  > "b")

  void <<< pure $ (true <= false)
  void <<< pure $ (1    <= 2)
  void <<< pure $ (1.0  <= 2.0)
  void <<< pure $ ('a'  <= 'b')
  void <<< pure $ ("a"  <= "b")

  void <<< pure $ (true >= false)
  void <<< pure $ (1    >= 2)
  void <<< pure $ (1.0  >= 2.0)
  void <<< pure $ ('a'  >= 'b')
  void <<< pure $ ("a"  >= "b")

  void <<< pure $ ("a"  <> "b")
  void <<< pure $ ("a"  `append` "b")

  void <<< pure $ (true && false)
  void <<< pure $ (true `conj` false)
  void <<< pure $ (true `disj` false)
  void <<< pure $ (not true)
  void <<< pure $ (not false)

  void <<< pure $ (1    + 2)
  void <<< pure $ (1.0  + 2.0)

  void <<< pure $ (1    `add` 2)
  void <<< pure $ (1.0  `add` 2.0)

  void <<< pure $ (1    - 2)
  void <<< pure $ (1.0  - 2.0)

  void <<< pure $ (1    `sub` 2)
  void <<< pure $ (1.0  `sub` 2.0)

  void <<< pure $ (1    * 2)
  void <<< pure $ (1.0  * 2.0)

  void <<< pure $ (1    `mul` 2)
  void <<< pure $ (1.0  `mul` 2.0)

  void <<< pure $ (1    / 2)
  void <<< pure $ (1.0  / 2.0)

  void <<< pure $ (1    `div` 2)
  void <<< pure $ (1.0  `div` 2.0)

  void <<< pure $ (1    `mod` 2)

  void <<< pure $ (-2.0)

  "hi" # pure >>> void

```

### CODE REVIEW

File: src/Language/PureScript/CodeGen/JS/Optimizer/Inliner.hs [hunk 1]
```diff
@@ -246,71 +239,130 @@ inlineFnComposition :: (Applicative m, MonadSupply m) => JS -> m JS
...
+opGreaterThanOrEq :: [(String, String)]
+opGreaterThanOrEq = [(C.prelude, (C.>=)), (C.dataOrd, C.greaterThanOrEq)]
 
-eqString :: (String, String)
-eqString = (C.prelude, C.eqNumber)
+opAppend :: [(String, String)]
+opAppend = [(C.prelude, (C.<>)), (C.prelude, (C.++)), (C.prelude, C.append), (C.dataSemigroup, C.append)]
 
-eqBoolean :: (String, String)
-eqBoolean = (C.prelude, C.eqNumber)
```
Comment (garyb) [inline src/Language/PureScript/CodeGen/JS/Optimizer/Inliner.hs]: Gee, I wonder why the boolean and string equality inlining wasn't working :scream:

Comment (paf31) [inline src/Language/PureScript/CodeGen/JS/Optimizer/Inliner.hs]: :fearful: We really should find a way to test this in Travis.

File: src/Language/PureScript/CodeGen/JS/Optimizer.hs [hunk 2]
```diff
@@ -79,7 +65,9 @@ optimize' js = do
     , inlineVariables
     , inlineValues
     , inlineOperator (C.prelude, (C.$)) $ \f x -> JSApp f [x]
```
Comment (paf31) [inline src/Language/PureScript/CodeGen/JS/Optimizer.hs]: Do these rules ever fire now?

Comment (garyb) [inline src/Language/PureScript/CodeGen/JS/Optimizer.hs]: Yeah, for people with the current prelude: `($)` is not an alias there.

**Comment by paf31** [comment]
:+1: Looks great.

**Comment by garyb** [comment]
Cool! Not sure if you saw my message on IRC, but once this is merged I'm happy to release another RC/initial 0.8, the rest of the stuff assigned to me can definitely wait for an .1 as far as I'm concerned.

**Pull request merged**